File: c++/268_Missing_Number.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
	int missingNumber(vector<int>& nums) {
		int n = nums.size();
		int sum = 0;
		for (int i = 0; i < n; i++){
			sum += nums[i];
		}
		return n*(n + 1) / 2 - sum;
	}
	
	int missingNumber2(vector<int>& nums) {
		int n = nums.size();
		int a = 1, b = 1;
		for (int i = 0; i < n; i++){
			a ^= nums[i];
		}
		for (int i = 0; i < n + 1; i++){
			b ^= i;
		}
		return a ^ b;
	}
};

int main(){
	Solution s;
	int temp_arr[3] = { 0, 1, 3 };
	vector<int> temp(temp_arr, temp_arr + 3);
	cout << s.missingNumber(temp);

	return 0;
}

File: C#/268_Missing_Number.cs
using System;

namespace LeetCode
{
    public class Solution
    {
        public int MissingNumber(int[] nums)
        {
            int n = nums.Length;
            int sum = 0;
            foreach(int num in nums)
            {
                sum += num;
            }
            return n * (n + 1) / 2 - sum;
        }

        public int MissingNumber2(int[] nums)
        {
            int a = 1, b = 1;
            foreach(int num in nums)
            {
                a ^= num;
            }
            for (int i = 0; i < nums.Length + 1; i++)
            {
                b ^= i;
            }
            return a ^ b;
        }

        //Testcase
        static void Main(string[] args)
        {
            Solution s = new Solution();
            Console.Write(s.MissingNumber(new int[] { 0, 1, 3 }));
        }
    }
}
